storage/google/cloud/storage/bucket.py

```diff
@@ -199,6 +200,47 @@ def create(self, client=None):
             data=properties, _target_object=self)
         self._set_properties(api_response)
 
+    def patch(self, client=None):
+        """Sends all changed properties in a PATCH request.
+
+        Updates the ``_properties`` with the response from the backend.
+
+        :type client: :class:`~google.cloud.storage.client.Client` or
+                      ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current object.
+        """
+        # Special case: For buckets, it is possible that labels are being
+        # removed; this requires special handling.
+        if self._label_removals:
+            self._changes.add('labels')
+            self._properties.setdefault('labels', {})
+            for removed_label in self._label_removals:
+                self._properties['labels'][removed_label] = None
+
+        # Call the superclass method.
+        answer = super(Bucket, self).patch(client=client)
+
+        # Clear out the label removals.
+        # This comes after the superclass method to ensure that we hold on
+        # to the data in case of an error.
+        self._label_removals.clear()
+        return answer
+
+    def update(self, client=None):
+        """Sends all properties in a PUT request.
+
+        Updates the ``_properties`` with the response from the backend.
+
+        :type client: :class:`~google.cloud.storage.client.Client` or
+                      ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current object.
+        """
+        answer = super(Bucket, self).update(client=client)
+        self._label_removals.clear()
```

Comment:
Clearing `self._label_removals` would be better done inside `_set_properties` (so that it get cleared during a `reload` too.